### PR TITLE
As a user, the game requests input from each player when it is their turn

### PR DESCRIPTION
**Pull request by ellehallal**
- Create player class. Can create a player with a mark
- Outputs the current player's mark
- Toggles the current player after a move is made
- Requests input from player when it is their turn

### CODE REVIEW

File: lib/controller.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 class Controller
+
   def initialize(game)
     @game = game
   end
 
+  def play_move
```
Comment (matthewglover) [inline lib/controller.rb]: This method calls two consecutive methods on game. Which suggests to me that this behaviour might belong in game.

File: spec/controller_spec.rb [hunk 2]
```diff
@@ -5,12 +5,24 @@
...
-  describe 'Controller: ' do
+  describe 'Controller: alternates current player after each move ' do
     board = Board.new([1, 2, 3, 4, 5, 6, 7, 8, 9])
     display = Display.new
     player1 = Player.new('x')
     player2 = Player.new('o')
     game = Game.new(board, display, player1, player2)
     controller = Controller.new(game)
+
+    it 'player1 makes a move then current player toggles to player2' do
```
Comment (matthewglover) [inline spec/controller_spec.rb]: Could use context blocks here to make the tests read more like:

context: playing a move when x is the current player, it plays x's move and updates current player.

File: lib/game.rb [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+  def initialize(board, display, player1, player2)
+    @board = board
+    @display = display
     @player1 = player1
     @player2 = player2
     @current_player = player1
   end
 
-  def prompt_make_move
+  def prompt_for_move
```
Comment (matthewglover) [inline lib/game.rb]: Should this method be private?

File: lib/game.rb [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
       @current_player = @player2
     else
       @current_player = @player1
     end
   end
+
+  def make_move
+    @display.display_board(@board.board)
+    @display.ask_for_move
+    move = prompt_for_move
```
Comment (matthewglover) [inline lib/game.rb]: What is the difference between asking for a move and prompting for a move?

File: spec/game_spec.rb [hunk 5]
```diff
@@ -1,29 +1,62 @@
...
-      game.toggle_player
+      board = Board.new([1, 2, 3, 4, 5, 6, 7, 8, 9])
+      game = Game.new(board, display, player1, player2)
+
+      game.toggle_current_player
+      game.toggle_current_player
       expect(game.current_player.mark).to eq('x')
     end
+
+    it 'prompts for moves and displays updated board' do
```
Comment (matthewglover) [inline spec/game_spec.rb]: "prompts for moves and displays the/an updated board"

File: lib/controller.rb (hunk 1 above)
Comment (ellehallal) [reply lib/controller.rb]: Thanks, Matt. I wasn't sure where this method should be placed. I'll move this into Game

File: lib/game.rb (hunk 4 above)
Comment (ellehallal) [reply lib/game.rb]: The difference is ask_for_move displays "Choose a move between 1-9" and prompt_for_move requests input. 
Maybe the prompt should be in ask_for_move, and then prompt_for_move will be redundant. What do you think?

File: spec/controller_spec.rb (hunk 2 above)
Comment (ellehallal) [reply spec/controller_spec.rb]: Your example is much clearer and descriptive. I'll add this, thanks Matt.

File: lib/game.rb (hunk 3 above)
Comment (ellehallal) [reply lib/game.rb]: Taking into considering the feedback below, I believe this method is redundant. However, I will consider making other methods private. Thank you.

File: spec/controller_spec.rb [hunk 6]
```diff
@@ -1,30 +1,16 @@
...
-     x | 2 | 3
-    -----------
-     4 | 5 | 6
-    -----------
-     7 | 8 | 9\n""")
-    .to_stdout
+    player1 = Player.new('x')
+    player2 = Player.new('o')
+    game = Game.new(board, display, player1, player2)
+    controller = Controller.new(game)
```
Comment (Kotauror) [inline spec/controller_spec.rb:14]: Is the Controller still needed in the current design?

**Pull request merged**